componentsv2/src/main/java/uk/gov/android/ui/componentsv2/topappbar/GdsTopAppBar.kt

```diff
@@ -0,0 +1,390 @@
+import uk.gov.android.ui.componentsv2.button.GdsIconButton
+import uk.gov.android.ui.componentsv2.button.GdsIconButtonDefaults
+import uk.gov.android.ui.componentsv2.button.IconButtonContent
+import uk.gov.android.ui.theme.m3.GdsLocalColorScheme
+import uk.gov.android.ui.theme.m3.GdsTheme
+import uk.gov.android.ui.theme.m3.Typography
+import uk.gov.android.ui.theme.util.UnstableDesignSystemAPI
+
+/**
+ * This is a TopAppBar that meets the GDS requirements and provides an already configured bar.
+ *
+ * @param modifier - modifier applied to the entire TopAppBar
+ * @param title - **optional** defaults to null - GdsHeading component with a Title 2 style. Can be displayed either Start or Centre, however it it always displayed between the navigation button and actions (if existing)
+ * @param navigationButton - **optional** defaults to Back icon button - the icon displayed on teh left side of the bar
+ * @param onClick - defaults to a void action - action for the navigation icon button and can be left as default if no navigation icon button has been provided or is null
+ * @param actions - **optional** defaults to null - Icon Buttons that will be displayed on the Start of the Bar
+ * @param menu - **optional** defaults to null - allows for a DropDownMenu to be passed in and displayed - it is passed in as a Composable and managed by the consumer
+ * @param alignment - defaults to Left - applies to the text only and manages if a [CenterAlignedTopAppBar] or a [TopAppBar] will be used
+ * @param scrollBehaviour - **optional** defaults to null - used if a scrolled behaviour is required (see [GdsTopAppBarDefaults.scrollBehaviour] for GDS default scroll behaviour)
+ * @param topAppBarColors - default to [GdsTopAppBarDefaults.colors] and can be overridden and/ or customised
+ */
+@OptIn(ExperimentalMaterial3Api::class, UnstableDesignSystemAPI::class)
+@Composable
+fun GdsTopAppBar(
+    modifier: Modifier = Modifier,
+    title: String? = null,
+    navigationButton: IconButtonContent? = GdsIconButtonDefaults.defaultBackContent(),
+    onClick: () -> Unit = {},
+    actions: ImmutableList<TopBarActionButton>? = null,
+    alignment: TopAppBarAlignment = TopAppBarAlignment.Start,
+    scrollBehaviour: TopAppBarScrollBehavior? = null,
+    topAppBarColors: TopAppBarColors = GdsTopAppBarDefaults.colors(),
+    menu: (@Composable () -> Unit)? = null,
+) {
+    when (alignment) {
+        TopAppBarAlignment.Start -> {
+            StartAlignedTopAppBar(
+                title,
+                onClick,
+                navigationButton,
+                topAppBarColors,
+                actions,
+                scrollBehaviour,
+                modifier,
+                menu,
+            )
+        }
+
+        TopAppBarAlignment.Centre -> {
+            CenterAlignedTopAppBar(
+                title = {
+                    title?.let {
+                        // This needs to be kept as test rather tha use the GdsHeading because the heading affects the
```

Comment:
Think you mean text here...?
```suggestion
                        // This needs to be kept as text rather tha use the GdsHeading because the heading affects the
```